Adding capability for charged particle motion in electric and magnetic fields (#20)
idaholab/salamander: doc/content/source/userobjects/LeapFrogStepper.md

> # LeapFrogStepper

Reviewer: Do you have references for this approach? Please add it to this page. You can also add your submitted paper on this page and the BorisStepper page, we'll just update it once it's published.

Author: There is some more detail about this in the Birdsall Langdon book and it has been added as a reference

!syntax description /UserObjects/LeapFrogStepper

> 

Reviewer: Could you please add more context here? Why is this method important? What is it used for?

Author: More context has been added

> On the first step that a particle takes the particle velocity is updated with a 1/2 time step

Reviewer: ```suggestion
On the first step that a particle takes the particle velocity is updated with a 1/2 time step as
```

\begin{equation}
  \vec{v}_{1/2} = \vec{v}_0 + \frac{q}{m} \vec{F} \frac{dt}{2}

> \end{equation}

Reviewer: please explain all the notations (see comments provided in `BorisStepper.md`)
```suggestion
\begin{equation}
  \vec{v}_{1/2} = \vec{v}_0 + \frac{q}{m} \vec{F} \frac{dt}{2},
\end{equation}
where ...
```

> The particle position is then updated with a full time step

Reviewer: ```suggestion
The particle position is then updated with a full time step as
```

> \begin{equation}
>   \vec{r}_{1} = \vec{r}_0 + \vec{v}_{1/2} dt
> \end{equation}

Reviewer: ```suggestion
\begin{equation}
  \vec{r}_{1} = \vec{r}_0 + \vec{v}_{1/2} dt,
\end{equation}
with ....
```

Reviewer: Again, please describe notations.

> For every other time step, n, after the initial both are updated with a full time step and

Reviewer: ```suggestion
For every other time step, $n$, after the initial both are updated with a full time step and
```
Are you missing a word in ` after the initial both` ?

\begin{equation}
  \vec{v}_{n + 1/2} = \vec{v}_{n - 1/2} + \frac{q}{m} \vec{F} dt
\end{equation}

\begin{equation}
  \vec{r}_{n} = \vec{r}_{n-1} + \vec{v}_{n + 1/2} dt

> \end{equation}

Reviewer: Make sure to incorporate these equations in a sentence and define the notations that have not yet been defined.

# LeapFrogStepper

!syntax description /UserObjects/LeapFrogStepper

> When charged particles are only subjected to an electric field a standard particle stepping scheme called the leapfrog scheme is used to accurately calculate the motion of these particles through the field. This scheme can repoduce particle paths exactly, to machine precision in the case of a constant uniform electric field. More information about this particle stepping scheme can be found in [!cite](Birdsall_Langdon_1991).

Reviewer: ```suggestion
When charged particles are only subjected to an electric field, a standard particle stepping scheme called the leapfrog scheme can be used to accurately calculate the motion of these particles through the field. This scheme can reproduce particle paths exactly, to machine precision, in the case of a constant uniform electric field. More information about this particle stepping scheme can be found in [!cite](Birdsall_Langdon_1991).
```

On the first step that a particle takes the particle velocity is updated with a 1/2 time step as

\begin{equation}
  \vec{v}_{1/2} = \vec{v}_0 + \frac{q}{m} \vec{F}_0 \frac{dt}{2}
\end{equation}

> Where $\vec{v}_{0}$ is the initial particle velocity, $q$, and $m$ are the particles charge and mass, $\vec{F}_0$ is the initial force field the particle is subject to, and $\vec{v}_{1/2}$ is the particle velocity 1/2 of a time step after the simulation start time.

Reviewer: ```suggestion
where $\vec{v}_{0}$ is the initial particle velocity, $q$, and $m$ are the particles charge and mass, $\vec{F}_0$ is the initial force field the particle is subject to, and $\vec{v}_{1/2}$ is the particle velocity 1/2 of a time step after the simulation start time.
```

The particle position is then updated with a full time step as

\begin{equation}
  \vec{r}_{1} = \vec{r}_0 + \vec{v}_{1/2} dt
\end{equation}

> Where $\vec{r}_{1}$ is the particles' position after the first time step has been taken and $\vec{r}_0$ is the initial position of the particle. On the $n^\text{th}$ step after the initial step both the particles' velocity and postition are updated with a full time step

Reviewer: ```suggestion
where $\vec{r}_{1}$ is the particles' position after the first time step has been taken and $\vec{r}_0$ is the initial position of the particle. On the $n^\text{th}$ step after the initial step both the particles' velocity and postition are updated with a full time step
```

\begin{equation}
  \vec{v}_{n + 1/2} = \vec{v}_{n - 1/2} + \frac{q}{m} \vec{F} dt
\end{equation}

> Where $\vec{v}_{n + 1/2}$ is the particles' velocity at a time which is 1/2 of a time step after the time at time step $n$, and $\vec{v}_{n - 1/2}$ is the particles' velocity at a time which is 1/2 of a time step before the time at time step $n$. Then the new particle position is updated using the velocity from 1/2 step in the future.

Reviewer: ```suggestion
where $\vec{v}_{n + 1/2}$ is the particles' velocity at a time which is 1/2 of a time step after the time at time step $n$, and $\vec{v}_{n - 1/2}$ is the particles' velocity at a time which is 1/2 of a time step before the time at time step $n$. Then the new particle position is updated using the velocity from 1/2 step in the future.
```

\begin{equation}
  \vec{r}_{n} = \vec{r}_{n-1} + \vec{v}_{n - 1/2} dt
\end{equation}

> Where $\vec{r}_{n}$ is the particles position at the time corrisponding to the $n^\text{th}$ time step and $\vec{r}_{n-1}$ is the particles position on the previous time step.

Reviewer: ```suggestion
where $\vec{r}_{n}$ is the particles position at the time corresponding to the $n^\text{th}$ time step and $\vec{r}_{n-1}$ is the particles position on the previous time step.
```

> The implementation of the Leap frog scheme was verified using single particle motion tests: parallel acceleration in an electric field ([/parallel_acceleration.i]), and projectile motion ([/projectile_motion.i]).

Reviewer: ```suggestion
The implementation of the leap frog scheme was verified using single particle motion tests: parallel acceleration in an electric field ([/parallel_acceleration.i]), and projectile motion ([/projectile_motion.i]).
```

Reviewer: Thanks for the references to the inputs!
